function_calling/src/function_calling.py

```diff
@@ -87,31 +87,33 @@ def set_llm(self, api: str):
             raise ValueError(f"Invalid LLM API: {api}, only'sambastudio' and'sambaverse' are supported.")
         return llm
 
-    def get_tools_schemas(self, tools: Union[Tool, list] = None, default: Union[Tool, BaseModel] = None):
-        if tools is None:
+    def get_tools_schemas(
+        self, tools: Optional[Union[Tool, list]] = None, default: Optional[Union[Tool, BaseModel]] = None
+    ):
+        if tools is None or isinstance(tools, list):
             pass
         elif isinstance(tools, Tool):
```

Comment:
This will break if tools is just one tool.

Reply:
fixed